src/static_canvas.class.js

```diff
@@ -871,16 +841,34 @@
         fabric.util.clipContext(this, canvasToDrawOn);
       }
 
-      this._renderBackground(canvasToDrawOn);
-      this._renderObjects(canvasToDrawOn, activeGroup);
-      this._renderActiveGroup(canvasToDrawOn, activeGroup);
+      if (activeGroup && !this.preserveObjectStacking) {
```

Comment:
I'm concerned that we're making code worse here. `_renderBackground`, `_renderObjects`, `_renderActiveGroup` are easy to read and understand because they're all on the same level of abstraction and are named well. You can then look into each of them for implementation details. This is the kind of stuff we should be doing MORE in our code but here I see that we're going the opposite way and introducing a relatively big chunk of some sort of procedure without naming it.

Yes — sometimes we DO need to "unroll" things, such as for performance. Is this why we're doing it here? Is there a way to keep same "style" as before but only change implementation?